packages/trace-viewer/src/traceModel.ts

```diff
@@ -47,50 +46,54 @@ export class TraceModel {
   }
 
   async load(traceURL: string, progress: (done: number, total: number) => void) {
-    this.contextEntry.traceUrl = traceURL;
     this._zipReader = new zipjs.ZipReader( // @ts-ignore
         new zipjs.HttpReader(this._formatUrl(traceURL), { mode: 'cors', preventHeadRequest: true }),
         { useWebWorkers: false }) as zip.ZipReader;
-    let traceEntry: zip.Entry | undefined;
-    let networkEntry: zip.Entry | undefined;
-    let stacksEntry: zip.Entry | undefined;
+
+    const ordinals: string[] = [];
+    let hasSource = false;
     for (const entry of await this._zipReader.getEntries({ onprogress: progress })) {
-      if (entry.filename.endsWith('.trace'))
-        traceEntry = entry;
-      if (entry.filename.endsWith('.network'))
-        networkEntry = entry;
-      if (entry.filename.endsWith('.stacks'))
-        stacksEntry = entry;
+      const match = entry.filename.match(/([\d]+-)?trace\.trace/);
+      if (match)
+        ordinals.push(match[1] || '');
       if (entry.filename.includes('src@'))
-        this.contextEntry.hasSource = true;
+        hasSource = true;
       this._entries.set(entry.filename, entry);
     }
-    if (!traceEntry)
+    if (!ordinals.length)
       throw new Error('Cannot find .trace file');
 
     this._snapshotStorage = new PersistentSnapshotStorage(this._entries);
 
-    const traceWriter = new zipjs.TextWriter() as zip.TextWriter;
-    await traceEntry.getData!(traceWriter);
-    for (const line of (await traceWriter.getData()).split('\n'))
-      this.appendEvent(line);
+    for (const ordinal of ordinals) {
+      const contextEntry = createEmptyContext();
+      contextEntry.traceUrl = traceURL;
+      contextEntry.hasSource = hasSource;
+
+      const traceWriter = new zipjs.TextWriter() as zip.TextWriter;
+      const traceEntry = this._entries.get(ordinal + 'trace.trace')!;
+      await traceEntry!.getData!(traceWriter);
+      for (const line of (await traceWriter.getData()).split('\n'))
+        this.appendEvent(contextEntry, line);
 
-    if (networkEntry) {
       const networkWriter = new zipjs.TextWriter();
-      await networkEntry.getData!(networkWriter);
+      const networkEntry = this._entries.get(ordinal + 'trace.network')!;
+      await networkEntry?.getData?.(networkWriter);
       for (const line of (await networkWriter.getData()).split('\n'))
-        this.appendEvent(line);
-    }
+        this.appendEvent(contextEntry, line);
 
-    if (stacksEntry) {
-      const writer = new zipjs.TextWriter();
-      await stacksEntry.getData!(writer);
-      const metadataMap = parseClientSideCallMetadata(JSON.parse(await writer.getData()));
-      for (const action of this.contextEntry.actions)
-        action.stack = action.stack || metadataMap.get(action.callId);
-    }
+      const stacksWriter = new zipjs.TextWriter();
+      const stacksEntry = this._entries.get(ordinal + 'trace.stacks')!;
+      if (stacksEntry) {
```

Comment:
nit: either null-check here or question mark on the previous line, but not both?